srcs/http/response/http_path_check/http_path_check.cpp

```diff
@@ -0,0 +1,107 @@
+#include "http_path_check.hpp"
+#include <cstdlib> // atoi
+#include <iostream>
+
+namespace http {
+
+CheckPathResult HttpPathCheck::Check(const MockDtoServerInfos &server_info, HttpRequest &request) {
+	CheckPathResult result;
+
+	CheckDTOServerInfo(result, server_info, request.header_fields);
+	CheckLocationList(result, server_info.locations, request.request_line.request_target);
+	return result;
+}
+
+// Check Server_info
+void HttpPathCheck::CheckDTOServerInfo(
+	CheckPathResult &result, const MockDtoServerInfos &server_info, HeaderFields &header_fields
+) {
+	if (server_info.host != header_fields["Host"]) { // Check host_name
+		result.is_ok = CheckPathResult::INVALID_HOST;
+	} else if (static_cast<size_t>(std::atoi(header_fields["Content-Length"].c_str())) >
+			   server_info.client_max_body_size) { // Check content_length
+		result.is_ok = CheckPathResult::PAYLOAD_TOO_LARGE;
+	} else if (!server_info.error_page.second.empty()) { // Check error_page
+		result.error_status_code = server_info.error_page.first;
+		result.error_page_path   = server_info.error_page.second;
+	}
+	return;
+}
+
+// Check LocationList
+void HttpPathCheck::CheckLocationList(
+	CheckPathResult &result, const LocationList &locations, const std::string &request_target
+) {
+	if (result.is_ok != CheckPathResult::OK) {
+		return;
+	}
+	MockLocationCon match_location = CheckLocation(result, locations, request_target);
```

Comment:
その後変わらなさそうなので const 参照で良いかも
```suggestion
	const MockLocationCon &match_location = CheckLocation(result, locations, request_target);
```